[18.0][MIG] mail_activity_done: Migration to 18.0

StefanRijnhart opened the pull request:
Continued from https://github.com/OCA/mail/pull/17

The functionality is now supported by Odoo, using a similar datamodel: to mark activities done, they are archived instead of unlinked. This is reflected in the way that the state field is computed.

We can keep this module to ensure that all activity types default to keeping done activities (which is not the case in Odoo). Otherwise, the migration script will also clear archived activities that were not done, to prevent them from showing up as 'done' in Odoo 18.

JordiBForgeFlow committed:
[11.0][mail_activity_done] add new module to keep activities that
have been completed.

fixup moment js

fixup

AdriaGForgeFlow committed:
[MIG] Migrate module mail_activity_done to v12

MiquelRForgeFlow committed:
[FIX] mail_activity_done: several fixes

AdriaGForgeFlow committed:
[12.0][FIX] mail_activity_done module

AdriaGForgeFlow committed:
[FIX] search view

AdriaGForgeFlow committed:
[ADD] uninstall_hook to remove done activities

AdriaGForgeFlow committed:
[UPD] Version number

JordiBForgeFlow committed:
[mail_activity_done] set activities as inactive when they have been completed.

manuelregidor committed:
[IMP]mail_activity_done: black, isort, prettier

manuelregidor committed:
[MIG]mail_activity_done: Migration to 14.0

BernatPForgeFlow committed:
[IMP] mail_activity_done: black, isort, prettier

BernatPForgeFlow committed:
[15.0][MIG] mail_activity_done: Migration to 15.0

Commit:
[IMP] mail_activity_done: pre-commit stuff

Commit:
[MIG] mail_activity_done: Migration to 16.0

Changed property in test to a correct one

Changed README

Added necessary checks if module does not have an icon

Added default value to _original_module

hbrunn committed:
[FIX] mail_activity_done: patch _action_done in a more reliable way

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (15 of 15 strings)

Translation: social-16.0/social-16.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_done/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (15 of 15 strings)

Translation: social-16.0/social-16.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_done/it/

mlaitinen committed:
Added translation using Weblate (Finnish)

mlaitinen committed:
Translated using Weblate (Finnish)

Currently translated at 46.6% (7 of 15 strings)

Translation: social-16.0/social-16.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_done/fi/

mlaitinen committed:
Translated using Weblate (Finnish)

Currently translated at 60.0% (9 of 15 strings)

Translation: social-16.0/social-16.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_done/fi/

mlaitinen committed:
Translated using Weblate (Finnish)

Currently translated at 100.0% (15 of 15 strings)

Translation: social-16.0/social-16.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_done/fi/

Shamsidinkhon committed:
[IMP] mail_activity_done: exclude completed activities from progress bar

houzefa-abba committed:
Added translation using Weblate (French)

houzefa-abba committed:
Translated using Weblate (French)

Currently translated at 100.0% (15 of 15 strings)

Translation: social-16.0/social-16.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-16-0/social-16-0-mail_activity_done/fr/

OCA-git-bot committed:
[BOT] post-merge updates

vvrossem committed:
[IMP] mail_activity_done: black, isort, prettier

vvrossem committed:
[MIG] mail_activity_done: Migration to 17.0

vvrossem committed:
[FIX] mail_activity_done: Remove broken and apparently unused js file

From [[15.0][FIX] mail_activity_done: Remove broken and apparently unused js file OCA#994](OCA#994)

Commit:
[UPD] Update mail_activity_done.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update mail_activity_done.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-17.0/social-17.0-mail_activity_done
Translate-URL: https://translation.odoo-community.org/projects/social-17-0/social-17-0-mail_activity_done/

achulii committed:
[IMP] mail_activity_done: pre-commit auto fixes

Bart-dh commented:
Isn't this default functionality in Odoo 18?
<img width="240" alt="image" src="https://github.com/user-attachments/assets/a45876e8-17c6-4f76-bfb8-aef7485a0e05" />

StefanRijnhart commented:
@Bart-dh Ooooh, nice find! I'll check if there is remaining functionality to keep, and if migration is required for done activities from earlier versions.

StefanRijnhart commented:
@Bart-dh Thanks again! Indeed, Odoo 18 supports keeping done activities natively. I refactored the module to smoothen out the migration, and to ensure that the activity types in Odoo are always configured to keep done activities.

cc @hbrunn

hbrunn commented on mail_activity_done/migrations/18.0.1.0.0/post-migration.py:
```diff
@@ -0,0 +1,11 @@
+    _set_keep_done(cr)
+    cr.execute(
+        """
+        delete from mail_activity where active is not true and done is not true;
```
this one is unexpected for me, what's your rationale here?

StefanRijnhart replied:
All archived activites are considered done in Odoo 18: https://github.com/odoo/odoo/blob/44e61c6/addons/mail/models/mail_activity.py#L154

hbrunn replied:
yes but we're migrating the module that people install to keep activities, so I'd not expect its migration to delete anything. I'd expect the migration script to set active to False where done is True (even though the v17 code of this module should have set this in the first place)

StefanRijnhart replied:
True, handling done activities that are still active is another inconsistency that might need handling. As for the other case, I downgraded that to a warning.

StefanRijnhart committed:
[MIG] mail_activity_done: Migration to 18.0

The functionality is now supported by Odoo, using a similar datamodel: to mark
activities done, they are archived instead of unlinked. This is reflected in the
way that the state field is computed.

hbrunn commented:
/ocabot merge nobump
